architecture-decision-records/adr-018-Knative_based_Kyma_eventing.md

```diff
@@ -0,0 +1,74 @@
+
+#### Architecture 
+
+
+                                +------------------+     +-----------------+
+                                |                  |     |                 |
+                                |  EventActivation |     |  Subscription   <---------+
+                                |                  |     |                 |         |
+                                +---------+--------+     +-----------+-----+         |
+                                          |                          |               |
+                                          |                          |               |
+                +---------------+         |    +---------------+     |               |
+                |               |     Read|    |               |     |Read/Update    |
+                |               |         |    |               |     |               |
+                |               |         |    |               |     |               |
+                |    Publish    |         +---->  Events Bus   <-----+               |
+                |               |              |  Controller   |                     |Create/Delete
+                |               |              |               |                     |
+                +--------+------+              +--+-------+----+                     |
+                         |                        |       |Create/Delete             |
+                         |Publish   Create/Delete |       |                          |
+                         |                        |       +-----+                    |
+                     +---v------------------------v----+        |             +------+-----+
+                     |                                 |        |             |            |
+                     |                                 +--------v-------------+  Subscriber|
+                     |         KN-Channel              |    KN-Subscription   |            |
+                     |                                 +----------------------+            |
+                     |                                 |                      +------------+
+              +------+---------------------------------+--------+
+              |                                                 |
+              |                                                 |
+              |                 Knative Bus                     |
+              |                                                 |
+              +-------------------------------------------------+
+
+### NATS Streaming
+
+Since knative provides an abstraction of "The Bus" which is a pluggable component of knative eventing that have different  implementations backed up by a certain messaging store "e.g. Kafka Bus, GCP PubSub Bus, etc..".
+
+### Decision
+
+1. Implementing a production ready `NATS Streaming` knative bus implementation.
```

Comment:
``1. Implement a production-ready Knative Bus `NATS Streaming`.``
``2. Provide `NATS Streaming` operator with `NATS Streaming` clusters which satisfy all production needs.`` -> not sure whether it's technically correct, though 🤷‍♀️ 
``3. Implement Knative Bus `Azure Service Bus Messaging` as a plan B.``
`4. Contribute both implementations to Knative Eventing.`